include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -0,0 +1,900 @@
+
+		const auto scaledKernelX = blit_utils_t::constructScaledKernel(kernelX, inExtent, outExtent);
+		const auto scaledKernelY = blit_utils_t::constructScaledKernel(kernelY, inExtent, outExtent);
+		const auto scaledKernelZ = blit_utils_t::constructScaledKernel(kernelZ, inExtent, outExtent);
+
+		std::ostringstream shaderSourceStream;
+		shaderSourceStream
+			<< "#version 460 core\n"
+			<< "#define _NBL_GLSL_WORKGROUP_SIZE_X_ " << workgroupSize << "\n"
+			<< "#define _NBL_GLSL_WORKGROUP_SIZE_Y_ " << 1 << "\n"
+			<< "#define _NBL_GLSL_WORKGROUP_SIZE_Z_ " << 1 << "\n"
+			<< "#define _NBL_GLSL_BLIT_DIM_COUNT_ " << static_cast<uint32_t>(imageType) + 1 << "\n"
+			<< "#define _NBL_GLSL_BLIT_ALPHA_BIN_COUNT_ " << paddedAlphaBinCount << "\n";
+
+		const auto castedFormat = getOutImageViewFormat(outFormat);
+		const uint32_t outChannelCount = asset::getFormatChannelCount(outFormat);
+		// Todo(achal): All this belongs in validation
+		{
+			const uint32_t inChannelCount = asset::getFormatChannelCount(inFormat);
+			assert(outChannelCount <= inChannelCount);
+
+			// inFormat should support SAMPLED_BIT format feature
+		}
+		const char* glslFormatQualifier = asset::IGLSLCompiler::getStorageImageFormatQualifier(castedFormat);
+
+		shaderSourceStream
+			<< "#define _NBL_GLSL_BLIT_OUT_CHANNEL_COUNT_ " << outChannelCount << "\n"
+			<< "#define _NBL_GLSL_BLIT_OUT_IMAGE_FORMAT_ " << glslFormatQualifier << "\n";
+
+		const core::vectorSIMDf scale = static_cast<core::vectorSIMDf>(inExtent).preciseDivision(static_cast<core::vectorSIMDf>(outExtent));
+
+		const core::vectorSIMDf minSupport(-scaledKernelX.negative_support[0], -scaledKernelY.negative_support[1], -scaledKernelZ.negative_support[2]);
+		const core::vectorSIMDf maxSupport(scaledKernelX.positive_support[0], scaledKernelY.positive_support[1], scaledKernelZ.positive_support[2]);
+
+		const uint32_t smemFloatCount = m_availableSharedMemory/(sizeof(float)*outChannelCount);
+		shaderSourceStream << "#define _NBL_GLSL_BLIT_SMEM_FLOAT_COUNT_ " << smemFloatCount << "\n";
+
+		if (alphaSemantic == asset::IBlitUtilities::EAS_REFERENCE_OR_COVERAGE)
+			shaderSourceStream << "#define _NBL_GLSL_BLIT_COVERAGE_SEMANTIC_\n";
+		if (outFormat != castedFormat)
+			shaderSourceStream << "#define _NBL_GLSL_BLIT_SOFTWARE_ENCODE_FORMAT_ " << outFormat << "\n";
+		shaderSourceStream << "#include <nbl/builtin/glsl/blit/default_compute_blit.comp>\n";
+
+		auto cpuShader = core::make_smart_refctd_ptr<asset::ICPUShader>(shaderSourceStream.str().c_str(), asset::IShader::ESS_COMPUTE, "CComputeBlit::createBlitSpecializedShader");
+		auto gpuUnspecShader = m_device->createShader(std::move(cpuShader));
+		auto specShader = m_device->createSpecializedShader(gpuUnspecShader.get(), { nullptr, nullptr, "main" });
+
+		return specShader;
+	}
+
+	template <typename KernelX, typename KernelY, typename KernelZ>
+	core::smart_refctd_ptr<video::IGPUComputePipeline> getBlitPipeline(
+		const asset::E_FORMAT inFormat,
```

Comment:
`inFormat` argument not needed.

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/343/files#r906733829

Reply:
Resolved.